xbmc/platform/darwin/DarwinUtils.mm

```diff
@@ -595,6 +604,150 @@ bool CFStringRefToStringWithEncoding(CFStringRef source, std::string &destinatio
   return manufName;
 }
 
+const std::vector<std::string>& CDarwinUtils::GetNameServers(void)
+{
+  static std::vector<std::string> nameServers;
+#if defined(TARGET_DARWIN_IOS)
+  if (nameServers.empty())
+  {
+    res_state res = static_cast<res_state>(malloc(sizeof(struct __res_state)));
+    
+    int result = res_ninit(res);
+    
+    if ( result == 0 )
+    {
+      for ( int i = 0; i < res->nscount; i++ )
+      {
+        std::string s = inet_ntoa(res->nsaddr_list[i].sin_addr);
+        nameServers.push_back(s);
+      }
+    }
+    else
+    {
+      CLog::Log(LOGERROR, "CDarwinUtils::GetNameServers - no nameservers could be fetched (error %d)", result);
+    }
+    
+    free(res);
+  }
+#endif
+  return nameServers;
+}
+
+#if defined(TARGET_DARWIN_IOS)
+const struct sockaddr *GetSockAddr(const struct rt_msghdr2 *rtm)
+{
+  //sockaddrs are after the message header
+  const struct sockaddr* dst_sa = reinterpret_cast<const struct sockaddr *>(rtm + 1);
+  
+  if(rtm->rtm_addrs & RTA_DST)
+  {
+    if(dst_sa->sa_family == AF_INET &&
+       !((rtm->rtm_flags & RTF_WASCLONED) &&
+         (rtm->rtm_parentflags & RTF_PRCLONING)))
+    {
+      if (dst_sa[RTAX_GATEWAY].sa_family == AF_INET)
+      {
+        return &dst_sa[RTAX_GATEWAY];
+      }
+    }
+  }
+  return nullptr;
+}
+
+bool GetGatewayForIfName(std::string ifName, struct sockaddr **foundSockAddr)
+{
+  bool result = false;
+  size_t needed = 0;
+  int mib[6];
+  char *buf = nullptr;
+  char *next = nullptr;
+  char *lim = nullptr;;
+  struct rt_msghdr2 *rtm = nullptr;
+  
+  mib[0] = CTL_NET;
+  mib[1] = PF_ROUTE;
+  mib[2] = 0;
+  mib[3] = 0;
+  mib[4] = NET_RT_DUMP2;
+  mib[5] = 0;
+  
+  if (sysctl(mib, 6, NULL, &needed, NULL, 0) < 0)
+  {
+    CLog::Log(LOGERROR, "sysctl: net.route.0.0.dump estimate");
+    return result;
+  }
+  
+  if ((buf = new char[needed]) == 0)
+  {
+    CLog::Log(LOGERROR, "malloc(%lu)", (unsigned long)needed);
+    return result;
+  }
+  if (sysctl(mib, 6, buf, &needed, NULL, 0) < 0)
+  {
+    CLog::Log(LOGERROR, "sysctl: net.route.0.0.dump");
+    delete [] buf;
+    return result;
+  }
+  
+  lim  = buf + needed;
+  
+  for (next = buf; next < lim; next += rtm->rtm_msglen)
+  {
+    rtm = (struct rt_msghdr2 *)next;
+    char nameBuff[IF_NAMESIZE];
+    std::string name = if_indextoname(rtm->rtm_index, nameBuff);
+    int flagVal = 1 << RTAX_GATEWAY;
+    
+    if (name == ifName && rtm->rtm_addrs & flagVal)
+    {
+      const struct sockaddr *addr = GetSockAddr(rtm);
+      if (addr != nullptr)
+      {
+        *foundSockAddr = const_cast<struct sockaddr *>(addr);
+        result = true;
+        break;
+      }
+    }
+  }
+  delete [] buf;
+  
+  return result;
+}
+#endif
+
+const std::string& CDarwinUtils::GetDefaultGateway(const std::string& ifName)
+{
+  static std::string gateway;
+  static std::string oldIfName = ifName;
+#if defined(TARGET_DARWIN_IOS)
+  if (gateway.empty() || oldIfName != ifName)
```

Comment:
still doesn't change the return value, if gateway for the same interface changes later